lotemplate/classes.py

```diff
@@ -77,6 +78,31 @@ def restart(self) -> None:
         self.__init__(self.host, self.port)
 
 
+class IfStatement:
+    """
+    Class representing an if statement in a template libreoffice
+    """
+    start_regex = r'\[\s*if\s*\$(\w+(\(.+?\))?)\s*(\=\=|\!\=)\s*([\w\-\.\,]+)\s*\]'
+    end_regex = r'(\[\s*endif\s*\])$'
```

Comment:
Here I prefer to keep my regex for two reasons

* first I need the length of the found string, so I need the parenthesis
* then this regex is used a lot (char by char between the if and the endif). I believe it will be much faster with the `$` at the end because the regex engine doesn't have to parse the entire string

Reply:
1. You don't need to create a matching group to get the whole match. The match group `0` already represents it.
2. Since regex engines starts from left to right, the speed/behavior won't change if you add a `$` at the end (apart from the fact that the pattern to match should be at the end of the line, of course). It would, however, change if you add a  `^` at the start of the regex, preventing the engine to search for the pattern if the first character of the line isn't `[`. But, in the end, it doesn't affect the speed of the program at all, as regex searches are already fast, and that the speed losses are mainly due to the opening/writing of the file, as well as the connection to the UNO bridge.

Reply:
ok, fixed